### PR TITLE
Add support for --ci

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Follow up: https://github.com/aspnet/CORS/pull/157

File: files/KoreBuild/KoreBuild.sh [hunk 1]
```diff
@@ -19,6 +19,12 @@ set_korebuildsettings() {
     return 0
 }
 
+set_korebuildcisettings() {
+    export DOTNET_CLI_TELEMETRY_OPTOUT=true
```
Comment (natemcmaster) [inline files/KoreBuild/KoreBuild.sh]: Can we add these too?

```
NUGET_PACKAGES=$(repo_root)/.nuget/packages
CI=true
DOTNET_HOME=$(repo_root)/.dotnet
```

File: scripts/bootstrapper/run.sh [hunk 2]
```diff
@@ -237,4 +239,6 @@ fi
 
 get_korebuild
 set_korebuildsettings "$tools_source" "$DOTNET_HOME" "$repo_path" "$config_file"
```
Comment (natemcmaster) [inline scripts/bootstrapper/run.sh]: Did you consider making $ci a new parameter on set_korebuildsettings?

File: files/KoreBuild/KoreBuild.sh (hunk 1 above)
Comment (pranavkm) [reply files/KoreBuild/KoreBuild.sh]: Sure

**Comment by pranavkm** [comment]
🆙 📅

File: files/KoreBuild/scripts/KoreBuild.psm1 [hunk 3]
```diff
@@ -308,10 +309,21 @@ function Set-KoreBuildSettings(
...
     if (!$ToolsSource) { $ToolsSource = 'https://aspnetcore.blob.core.windows.net/buildtools' }
 
+    if ($CI) {
+        $DotNetHome = Join-Path $RepoPath ".dotnet"
+
+        $env:CI = 'true'
+        $env:DOTNET_CLI_TELEMETRY_OPTOUT = 'true'
+        $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE = 'true'
+        $env:NUGET_SHOW_STACK = 'true'
+        $env:NUGET_PACKAGES = Join-Path $RepoPath '.nuget'
```
Comment (natemcmaster) [inline files/KoreBuild/scripts/KoreBuild.psm1]: Nit: for consistency with Korebuild.sh, `Join-Paths $RepoPath ('.nuget', 'packages')`

**Comment by davidfowl** [comment]
@pranavkm  This needs to be in release/2.1 as well right?

**Pull request merged**